lls/cache.c

```diff
@@ -399,6 +399,84 @@ xmit_icmp_reply(struct gatekeeper_if *iface, struct rte_mbuf *pkt)
 	return 0;
 }
 
+static void
+process_icmp_ping_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct lls_config *lls_conf, struct gatekeeper_if *iface)
+{
+	uint16_t tx_queue;
+	struct token_bucket_ratelimit_state *rs;
+	unsigned int i;
+	unsigned int num_granted_pkts;
+	unsigned int num_reply = 0;
+
+	if (iface == &lls_conf->net->front) {
+		tx_queue = lls_conf->tx_queue_front;
+		rs = &lls_conf->front_icmp_rs;
+	} else {
+		tx_queue = lls_conf->tx_queue_back;
+		rs = &lls_conf->back_icmp_rs;
+	}
+
+	num_granted_pkts = tb_ratelimit_allow_n(num_pkts, rs);
+
+	for (i = 0; i < num_pkts; i++) {
+		if (num_reply >= num_granted_pkts ||
+				xmit_icmp_ping_reply(iface, pkts[i]) != 0) {
```

Comment:
Is `rte_pktmbuf_free_bulk()` available in our DPDK? Once `xmit_icmp_ping_reply()` returns `void` (see another comment), we can just call `rte_pktmbuf_free_bulk()` here.

If `rte_pktmbuf_free_bulk()` is not available, could you open an issue on the new milestone "DPDK coordination", and add a `XXX` comment here to reference the new issue, so we can revisit this change once our DPDK is updated?